Assets/AppCenter/Plugins/AppCenterSDK/Auth/iOS/AuthInternal.cs

```diff
@@ -3,14 +3,38 @@
 
 #if UNITY_IOS && !UNITY_EDITOR
 using System;
-using System.Linq;
 using System.Collections.Generic;
 using System.Runtime.InteropServices;
+using Microsoft.AppCenter.Unity.Internal.Utility;
+using AOT;
 
 namespace Microsoft.AppCenter.Unity.Auth.Internal
 {
     class AuthInternal
     {
+
+#if ENABLE_IL2CPP
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+#endif
+        public delegate void NativeSignInCompletionHandler(IntPtr userInformation, IntPtr error);
+        private static AppCenterTask<UserInformation> _signInTask;
+
+        [MonoPInvokeCallback(typeof(NativeSignInCompletionHandler))]
+        public static void SignInCompletionHandlerNativeFunc(IntPtr userInformationPtr, IntPtr nsErrorPtr)
+        {
+            if (_signInTask != null)
```

Comment:
Can we set at least some flag here, and if it's already **true**, do nothing?

Reply:
I think it's the simplest approach. We can't really create the task instance here cause we need to return it.